src/services/lnd/onchain-service.ts

```diff
@@ -24,21 +24,17 @@ export const OnChainService = (
   }): Promise<SubmittedTransaction[] | OnChainServiceError> => {
     try {
       const { current_block_height } = await getHeight({ lnd })
-      const after = Math.max(0, current_block_height - scanDepth) // this is necessary for tests, otherwise after may be negative
-      const { transactions } = await getChainTransactions({ lnd, after })
 
-      return transactions
-        .filter((tx) => !tx.is_outgoing || !!tx.transaction)
-        .map((tx) => {
-          return {
-            confirmations: tx.confirmation_count || 0,
-            fee: toSats(tx.fee as number),
-            id: tx.id as TxId,
-            outputAddresses: tx.output_addresses as OnChainAddress[],
-            tokens: toSats(tx.tokens),
-            rawTx: decoder.decode(tx.transaction as string),
-          } as SubmittedTransaction
-        })
+      // this is necessary for tests, otherwise after may be negative
+      const after = Math.max(0, current_block_height - scanDepth)
+
+      return extractIncomingTransactions(
+        decoder,
+        await getChainTransactions({
```

Comment:
Minor nitpick -> Making the async function call here makes the code less readable than making the call on a separate line, and passing the result here